src/app/subscribe-events.ts

```diff
@@ -112,17 +137,27 @@ export function subscribeToComponentEvents(wrapperCtx: any) {
             let showInteraction = false;
             let highlightQuery: any = undefined;
 
+            const proteinId = wrapperCtx.proteinId;
+            const start = parseInt(e.detail.start);
+            const end = parseInt(e.detail.end);
+
             // Create query object from event data
-            if(e.detail.start && e.detail.end){
-                highlightQuery = {
-                    start_residue_number: parseInt(e.detail.start),
-                    end_residue_number: parseInt(e.detail.end)
-                };
+            if (start && end) {
+                highlightQuery = proteinId
+                    ? {
+                          uniprot_accession: proteinId,
+                          start_uniprot_residue_number: start,
+                          end_uniprot_residue_number: end,
+                      }
+                    : {
+                          start_auth_residue_number: start,
+                          end_auth_residue_number: end,
+                      };
```

Comment:
This block is exactly the same than before. Unless it's accidental repetition, we can abstract to function.